suite-native/biometrics/src/components/AuthenticatorProvider.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { ReactNode, useEffect } from 'react';
+import { AppState, Platform } from 'react-native';
+
+import * as LocalAuthentication from 'expo-local-authentication';
+
+import { BiometricOverlay } from './BiometricOverlay';
+import { useIsUserAuthenticated, useIsBiometricsEnabled } from '../biometricsAtoms';
+
+type AuthenticatorProviderProps = {
+    children: ReactNode;
+};
+
+export const AuthenticatorProvider = ({ children }: AuthenticatorProviderProps) => {
+    const { isBiometricsOptionEnabled } = useIsBiometricsEnabled();
+    const { isUserAuthenticated, setIsUserAuthenticated } = useIsUserAuthenticated();
+
+    useEffect(() => {
+        const authenticate = async () => {
+            if (Platform.OS === 'android') {
+                // Cancel any previous authentication attempt.
+                await LocalAuthentication.cancelAuthenticate();
+            }
+
+            const result = await LocalAuthentication.authenticateAsync();
+            setIsUserAuthenticated(result.success);
+        };
+
+        if (isBiometricsOptionEnabled && !isUserAuthenticated) {
```

Comment:
I would split this if and subsription lower to two separeted useEffects.

Reply:
https://github.com/trezor/trezor-suite/pull/8540/commits/b0340b2b1efdda51d31bc3a8bdd0dd4c36cbdda6